code/modules/power/singularity/emitter.dm

```diff
@@ -80,19 +97,19 @@
 		welded = FALSE
 
 /obj/machinery/power/emitter/RefreshParts()
-	var/max_firedelay = 120
-	var/firedelay = 120
-	var/min_firedelay = 24
+	var/max_fire_delay = 12 SECONDS
+	var/fire_shoot_delay = 12 SECONDS
+	var/min_fire_delay = 2.4 SECONDS
 	var/power_usage = 350
-	for(var/obj/item/stock_parts/micro_laser/L in component_parts)
-		max_firedelay -= 20 * L.rating
-		min_firedelay -= 4 * L.rating
-		firedelay -= 20 * L.rating
-	maximum_fire_delay = max_firedelay
-	minimum_fire_delay = min_firedelay
-	fire_delay = firedelay
-	for(var/obj/item/stock_parts/manipulator/M in component_parts)
-		power_usage -= 50 * M.rating
+	for(var/obj/item/stock_parts/micro_laser/laser in component_parts)
+		max_fire_delay -= 2 SECONDS * laser.rating
+		min_fire_delay -= 4 SECONDS * laser.rating
+		fire_shoot_delay -= 2 SECONDS * laser.rating
+	maximum_fire_delay = max_fire_delay
+	minimum_fire_delay = min_fire_delay
+	fire_delay = fire_shoot_delay
+	for(var/obj/item/stock_parts/manipulator/manipulator in component_parts)
+		power_usage -= 50 * manipulator.rating
```

Comment:
Shouldn't lasers effect the min fire delay in increments of 0.4 seconds rather then 4 seconds?

Reply:
indeed, i'll fix it